tests/foreman/cli/test_repository.py

```diff
@@ -397,6 +547,51 @@ def test_negative_create_with_auth_url_too_long(self):
                 with self.assertRaises(CLIFactoryError):
                     self._make_repository({u'url': url})
 
+    @tier1
+    def test_negative_create_with_invalid_download_policy(self):
+        """Verify that YUM repository cannot be created with invalid download policy
+
+        @id: 3b143bf8-7056-4c94-910d-69a451071f26
+
+        @Assert: YUM repository is not created with invalid download policy
+        """
+        with self.assertRaises(CLIFactoryError):
+            self._make_repository({
+                u'content-type': u'yum',
+                u'download-policy': gen_string('alpha', 5)
+            })
+
+    @tier1
+    def test_negative_update_to_invalid_download_policy(self):
+        """Verify that YUM repository cannot be updated to invalid download policy
+
+        @id: 5bd6a2e4-7ff0-42ac-825a-6b2a2f687c89
+
+        @Assert: YUM repository is not updated to invalid download policy
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            new_repo = self._make_repository({u'content-type': u'yum'})
+            Repository.update({
+                u'id': new_repo['id'],
+                u'download-policy': gen_string('alpha', 5)
+            })
+
+    @tier1
+    def test_negative_create_non_yum_with_download_policy(self):
+        """Verify that non-YUM repositories cannot be created with download policy
+
+        @id: 71388973-50ea-4a20-9406-0aca142014ca
+
+        @Assert: Non-YUM repository is not created with a download policy
+        """
+        for content_type in REPO_TYPE.keys():
```

Comment:
Good candidate to use `subTest` to make sure that all permutations are executed.